### DISCUSSION TITLE
Cocotb makefile to produce Coverage report from VCS

### DISCUSSION THREAD

**Discussion by Neom5** [cocotb/cocotb#3118] [Q&A]
How to write in makefile to produce Coverage report from VCS ?

In normal case (without using cocotb), make file goes like this :
```
SHELL		= /bin/bash
TEST_NAME	= !!!!!!.v # making testbench using SV
DESIGN_NAME	= !!!!!!!.v 

VCS		= vcs -full64 -debug_access+all ${DESIGN_NAME} ${TEST_NAME} -cm line+tgl+cond+fsm+branch
SIM_OPTS	= -l ./${TEST_NAME}.log  -cm line+tgl+cond+fsm+branch
SIM_GUI		= -l ./$(TEST_NAME).log -gui -cm line+tgl+cond+fsm+branch


vcs: 	 cmp sim rpt

vcs_gui: cmp sim_gui rpt

cmp:
	$(VCS)

sim:
	./simv $(SIM_OPTS)

sim_gui:
	./simv $(SIM_GUI)

rpt:
	urg -dir simv.vdb
```

**Comment by ktbarrett** [comment]
Add the options in `VCS` to the cocotb `COMPILE_ARGS` variable and the options in `SIM_OPTS` to the cocotb `SIM_ARGS` variable. Then you can use the cocotb `GUI` variable to turn decide to run in GUI mode or batch mode.

**Comment by aichichayedan** [comment]
Have you solved the problem?I had the same problem.
